src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -126,23 +134,33 @@ public static string GetFileName(string path)
             if (path == null)
                 return null;
 
+            return new string(GetFileName(path.AsReadOnlySpan()));
+        }
+
+        public static ReadOnlySpan<char> GetFileName(ReadOnlySpan<char> path)
+        {
             int offset = PathInternal.FindFileNameIndex(path);
             int count = path.Length - offset;
-            return path.Substring(offset, count);
+            return path.Slice(offset, count);
         }
 
         public static string GetFileNameWithoutExtension(string path)
         {
             if (path == null)
                 return null;
 
+            return new string(GetFileNameWithoutExtension(path.AsReadOnlySpan()));
```

Comment:
Again, check for length again. If you have the same length there is no need to new up another string.